src/org/openstreetmap/josm/gui/tagging/presets/TaggingPreset.java

```diff
@@ -419,23 +418,43 @@ public void applyComponentOrientation(ComponentOrientation o) {
             }
         };
         JPanel linkPanel = new JPanel(new GridBagLayout());
-        TaggingPresetItem previous = null;
+        List<PresetLink> alternativeTags = new ArrayList<>(); //list to store PresetLinks with alternative="true"
+        List<PresetLink> editAlsoTags = new ArrayList<>(); //list to store other presetLinks
         for (TaggingPresetItem i : data) {
             if (i instanceof Link) {
                 i.addToPanel(linkPanel, itemGuiSupport);
                 p.hasElements = true;
             } else {
                 if (i instanceof PresetLink) {
                     PresetLink link = (PresetLink) i;
-                    if (!(previous instanceof PresetLink && Objects.equals(((PresetLink) previous).text, link.text))) {
-                        itemPanel.add(link.createLabel(), GBC.eol().insets(0, 8, 0, 0));
+                    if (link.isAlternative()) {
+                        alternativeTags.add(link);
+                    } else {
+                        editAlsoTags.add(link);
                     }
                 }
-                if (i.addToPanel(itemPanel, itemGuiSupport)) {
+                if (!(i instanceof PresetLink) && (i.addToPanel(itemPanel, itemGuiSupport))) {
                     p.hasElements = true;
                 }
             }
-            previous = i;
+        }
+
+        if (!alternativeTags.isEmpty()) {
+            PresetLink link = new PresetLink();
+            itemPanel.add(link.createAlternativeLabel(), GBC.eol().insets(0, 8, 0, 0));
+            for (PresetLink links : alternativeTags) {
+                links.addToPanel(itemPanel, itemGuiSupport);
+                p.hasElements = true;
+            }
+        }
+
+        if (!editAlsoTags.isEmpty()) {
+            PresetLink link = new PresetLink();
+            itemPanel.add(link.createLabel(), GBC.eol().insets(0, 8, 0, 0));
+            for (PresetLink links : editAlsoTags) {
+                links.addToPanel(itemPanel, itemGuiSupport);
+                p.hasElements = true;
+            }
```

Comment:
If you consolidate the `createLabel`/`createAlternativeLabel`, this could be done like so:
```suggestion
        for (List<PresetLink> linkList : Arrays.asList(alternativeTags, editAlsoTags)) {
            if (!linkList.isEmpty()) {
                PresetLink link = new PresetLink();
                itemPanel.add(link.createLabel(), GBC.eol().insets(0, 8, 0, 0));
                for (PresetLink links : linkList) {
                    links.addToPanel(itemPanel, itemGuiSupport);
                    p.hasElements = true;
                }
            }
        }
```

Reply:
`link.isAlternative()` is always `false`, therefore same label is being created for both preset alternative and editalso tags. We may want to create label based on the `links` of the `linkedList`